src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java

```diff
@@ -94,6 +94,41 @@ public void validate(RequestedFlow flow, Set<String> bulkUpdateFlowIds)
         }
     }
 
+    /**
+     * Validates the specified flow.
+     *
+     * @param flow current flow state.
+     * @param requestedFlow a flow to be validated.
+     * @param bulkUpdateFlowIds flows to be ignored when check endpoints.
+     * @throws InvalidFlowException is thrown if a violation is found.
+     */
+    public void validate(Flow flow, RequestedFlow requestedFlow, Set<String> bulkUpdateFlowIds)
+            throws InvalidFlowException, UnavailableFlowEndpointException {
+        validate(requestedFlow, bulkUpdateFlowIds);
+        if (requestedFlow.getLoopSwitchId() != null) {
```

Comment:
Move it into some method. And you can make this check flow side/endpoint, not for whole flow. Just extend `EndpointDescriptor` with the required data.

Reply:
Moved to method. I don't think EndpointDescriptor is useful here.